### PR TITLE
Legacy invoice drift report 

### PR DESCRIPTION
**Pull request by ajeety4**
## Product Description

Internal, **read-only** reporting command for one time use - no UI and no data changes. It quantifies billing "drift" on legacy automated service-delivery invoices to inform the [CCCT-2525](https://dimagi.atlassian.net/browse/CCCT-2525) backfill decision.

### Summary of the problem

This report is meant to help identify whether any past invoices may have been generated with incorrect amounts. It is read-only and does not make any changes or corrections.

What is the issue?
The system allows duplicate deliveries to be recorded for the same completed work, and the payable amount is calculated based on the total number of deliveries.
For example, if a worker is paid 100 per delivery and the same delivery is recorded twice, the calculated amount for that completed work becomes 200.
Once a completed work has been invoiced, it is never billed again. However, duplicate deliveries can still be submitted afterwards. Since the work has already been invoiced, these additional deliveries are not paid.
This report helps identify invoices and completed works where duplicate deliveries may have affected the amount that was originally invoiced, so they can be reviewed and consider for correction if needed.
Noting that a this can also go vice versa - a duplicate delivery is being paid in the invoice but it was never agreed by PM. So invoice is overbilled. This is also being checked in the report

## Technical Summary

- Ticket: [CCCT-2524](https://dimagi.atlassian.net/browse/CCCT-2524)
- **Start here → [`report_invoice_drift.md`](https://github.com/dimagi/commcare-connect/blob/ay/invoice-drift-report/commcare_connect/opportunity/management/commands/report_invoice_drift.md)** — full logic, the two drift kinds, the `reconstruction_gap` confidence check, and the decision this informs.
- `manage.py report_invoice_drift` reconstructs the (pre-[CCCT-2505](https://dimagi.atlassian.net/browse/CCCT-2505)) billed count per invoice, compares it to today's agreement-gated count, and decomposes the difference into late-delivery / over-billing / org-pay drift. Writes two CSVs + a summary.
- 7 commits in build order (query → reconstruct → decompose → output → enrich → null-date edge case → docs); each is self-contained and green on its own.

Can be run from a management command (requires merging) or its equivalent on django shell.

### Safety story

Read-only: no writes, no migrations — only SELECTs, run manually via a management command or through shell. Inherently safe to existing data.

### Automated test coverage

21 tests in `test_invoice_drift.py` covering invoice selection, count reconstruction, each drift type, the `reconstruction_gap`, and the null-approval-date edge case.

### QA Plan

None

### Labels & Review

- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[CCCT-2524]: https://dimagi.atlassian.net/browse/CCCT-2524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CCCT-2525]: https://dimagi.atlassian.net/browse/CCCT-2525?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CCCT-2505]: https://dimagi.atlassian.net/browse/CCCT-2505?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: commcare_connect/opportunity/tests/test_invoice_drift.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+    _completed_work_with_visits(invoice, opportunity, amount=100, org_amount=0, agreed_in_window=1)
+
+    drift = compute_invoice_drift(invoice, base_url="https://staging.example.com/")
+    assert drift.reconstructed_flw_amount == Decimal(100)
+    assert drift.reconstruction_gap == Decimal(0)
+    assert drift.late_delivery_units == 0
+    assert drift.overbilled_units == 0
+    assert drift.org_pay_drift == Decimal(0)
+    assert drift.works[0].desired_deliveries_count == 1
+    assert drift.invoice_url.startswith("https://staging.example.com")
```
[the remark this answers is not in the document]
Comment (ajeety4) [reply commcare_connect/opportunity/tests/test_invoice_drift.py:216]: This is not an user facing input, so not an issue

File: commcare_connect/opportunity/management/commands/report_invoice_drift.py [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+                late_approved_visit_count=late_approved_visits,
+                late_delivery_drift=Decimal(late_units * full_rate),
+                overbilled_units=over_units,
+                overbilled_drift=Decimal(over_units * full_rate),
+                org_pay_drift=Decimal(reconstructed_deliveries_count * org_amount),
+                legacy_null_status_date_visits=null_date_visits,
+            )
+        )
+        reconstructed_flw_amount += reconstructed_deliveries_count * flw_amount
+
```
[the remark this answers is not in the document]
Comment (ajeety4) [reply commcare_connect/opportunity/management/commands/report_invoice_drift.py:333]: Noting that this is intentional as the child visits affects the parent completed work and the invocie.
Also it is just a metric , not used for calculation

**Review by sravfeyn — APPROVED** [review]
LGTM

**Comment by ajeety4** [comment]
> LGTM

Thanks Sravan for the review.
Note that I made a commit for db query optmisations, no logical changes. This is done for works without childs, which covers everything in Prod that we have today.

**Comment by ajeety4** [comment]
Closing this out as the report ran succesfully and the backfill strategy is decided.
